numpy/distutils/system_info.py

```diff
@@ -1766,23 +1849,11 @@ def _calc_info(self, name):
 
     def calc_info(self):
         user_order = os.environ.get(self.order_env_var_name, None)
-        if user_order is None:
-            lapack_order = self.lapack_order
-        else:
-            # the user has requested the order of the
-            # check they are all in the available list, a COMMA SEPARATED list
-            user_order = user_order.lower().split(',')
-            non_existing = []
-            lapack_order = []
-            for order in user_order:
-                if order in self.lapack_order:
-                    lapack_order.append(order)
-                elif len(order) > 0:
-                    non_existing.append(order)
-            if len(non_existing) > 0:
-                raise ValueError("lapack_opt_info user defined "
-                                 "LAPACK order has unacceptable "
-                                 "values: {}".format(non_existing))
+        lapack_order, _, unknown_order = _parse_order(self.lapack_order, user_order)
```

Comment:
Since this can throw, I'd recommend chaining some context to the exception
```suggestion
        try:
            lapack_order, _, unknown_order = _parse_order(self.lapack_order, user_order)
        except ValueError as e:
            raise ValueError("lapack_opt_info user defined LAPACK order unparseable") from e
```
or a better message of your choice.

Reply:
Alternatively the `_parse_order` could have an interface `def _parse_order(base_order, env_var)` which then can have `env_var` in the raised error? That would couple it to an env-var, directly, and also remove some boiler-plate code?

Reply:
changed to `_parse_env_order`, clarifies error message for end-users due to explicit env-var printing